### ISSUE TITLE
Youtube-dl configs and tubeup

### ISSUE THREAD

**Issue by vxbinaca** [bibanon/tubeup#4]
I'm getting youtube-dl from regular pip, not pip3. 

How do I set configs in the pip3 version so the script will respect that one?

My config is this:

```
-4 --download-archive ~/.ytdlarchive --retries 100 --no-overwrites --call-home --continue --write-info-json --write-description --write-thumbnail --write-annotations --all-subs --convert-subs srt --write-sub --add-metadata -f bestvideo+bestaudio/best --merge-output-format 'mkv' --prefer-ffmpeg --embed-thumbnail
```

Edit:

I tested this. Love the script but I have a few issues that prevent me from relying on it:

- Lack of subtitle ripping and uploading, compatible in a way that allows it's use on IAs video player automatically (they'd need to be in the SRT format, even though VTT seems to be common on youtube using ``` --convert-subs srt ``` would fix that).
- Not using ```-f bestvideo+bestaudio/best``` to dictate video quality, the first part is for youtube specifically, and the last ```/best``` is a catch-all for 99 percent of other sites out there.
- Perhaps consider switching to MKV as a video container? All the TV show rippers use it, it has maximum compatibility on OSs and the only slight downside is a slight delay in the video being streamable because of the derive task.

**Comment by antonizoon** [comment]
Hey there. We just used the default settings as given by the ArchiveTeam, but if you have better configs please do submit a pull request for us. 

One issue is that -f bestvideo+bestaudio/best had issues when i tried it before (especially outside of youtube), but if it works for you in youtube and out, I'll merge it.

**Comment by vxbinaca** [comment]
I would request you create a testing branch so that we can verify from multiple sources that presets work and so it doesn't pollute the main branch people rely on.

I tailored it for Youtube and Vimeo but it works on:

- Pornhub
- Vid.me
- Vine
- Tumblr and Tumblr embeds
- Motherless

The reason it works on 98 percent of sites is the ```/best```, because it just default to best when ripping it. Vimeo doesn't provide for ``` bestvideo+bestaudio``` and thats why it hung on that site, but Vimeo does allow ```best``` as does like 98 percent of other sites. All the ```/best``` does is give it an escape when it will fail for any site outside of youtube, in fact these -f lists get quite long on some peoples setups.

Another issue is the default use of avconv, and it's [outlined here](https://github.com/rg3/youtube-dl/issues/8622) why that's a terrible idea by the youtube-dl devs. I think ```--prefer-ffmpeg``` along with the removal of any mention of avconv in the README of this script is advised. A lot of my processing errors vanished once I forced FFMPEG use.

SRT subs is another one. Archive.org requires them to work. They can be ripped in VTT - which is what youtube seems to prefer - but should be converted to SRT with 

I'll get a PR ready once theres a testing branch. I will in detail carefully explain all the flags and why they should be used.
